### ISSUE TITLE
Just wondering what changes you have made since you managed to compile it

### ISSUE THREAD

**Issue by Question2** [fire-lizard/birthrt#1]
Hi,

Just wondering what changes you have managed to make after successfully compiling it? And how did you get it to compile exactly?

**Comment by Question2** [comment]
@fire-lizard

**Comment by fire-lizard** [comment]
Hi and thank you for interest in this excellent game!
To compile and especially **link** it I had to make literally tons of changes (see my commits named "Fixing link errors - part N").
In terms of compiling, first of all, I replaced old non compiled (in VS 2022) macros with more modern C++ code.
Also made many static inline methods just static (without inline).
At current stage, I am struggling with issues in MEMMANAG.CPP.
Please do not hesitate to ask me any other questions.

**Comment by seregavlg** [comment]
Hi, what compiler is needed to build? or you can post the compiled one?

**Comment by fire-waran** [comment]
I actually made it easy quite easy to compile the game, the only two things you will need are Visual Studio 2022 (don't forget to install C++ compiler and Windows SDK from the components, game does not need anything else), and Directx 9 SDK (from April 2005 or earlier, since it contains DirectPlay component). You may need to specify your own paths in there:
https://github.com/fire-lizard/birthrt/blob/main/CMakeLists.txt
Hope this helps. Please let me know if you still have difficulties to compile the game and I will help you.

**Comment by emanual4real** [comment]
Did you install Windows 10 or 11 SDK?  Is the C++ compiler C++ Clang Compiler for Windows (16.0.5) ?

**Comment by Question2** [comment]
> Hi and thank you for interest in this excellent game! To compile and especially **link** it I had to make literally tons of changes (see my commits named "Fixing link errors - part N"). In terms of compiling, first of all, I replaced old non compiled (in VS 2022) macros with more modern C++ code. Also made many static inline methods just static (without inline). At current stage, I am struggling with issues in MEMMANAG.CPP. Please do not hesitate to ask me any other questions.

@fire-lizard : just wondering if any progress has been made on this?

**Comment by fire-lizard** [comment]
> Did you install Windows 10 or 11 SDK? Is the C++ compiler C++ Clang Compiler for Windows (16.0.5) ?

No, only DirectX SDK. The game compiles both with MSVC and Clang Compiler

**Comment by fire-lizard** [comment]
> > Hi and thank you for interest in this excellent game! To compile and especially **link** it I had to make literally tons of changes (see my commits named "Fixing link errors - part N"). In terms of compiling, first of all, I replaced old non compiled (in VS 2022) macros with more modern C++ code. Also made many static inline methods just static (without inline). At current stage, I am struggling with issues in MEMMANAG.CPP. Please do not hesitate to ask me any other questions.
> 
> @fire-lizard : just wondering if any progress has been made on this?

Unfortunately not, don't know how to make MEMMANAG.CPP work in the modern OSes

**Comment by RodionKarimov** [comment]
> > > Hi and thank you for interest in this excellent game! To compile and especially **link** it I had to make literally tons of changes (see my commits named "Fixing link errors - part N"). In terms of compiling, first of all, I replaced old non compiled (in VS 2022) macros with more modern C++ code. Also made many static inline methods just static (without inline). At current stage, I am struggling with issues in MEMMANAG.CPP. Please do not hesitate to ask me any other questions.
> > 
> > 
> > [@fire-lizard](https://github.com/fire-lizard) : just wondering if any progress has been made on this?
> 
> Unfortunately not, don't know how to make MEMMANAG.CPP work in the modern OSes

There is some assembler code in sources and some nasty memory management, which can be hard to transfer to modern code base.
Why don't you continue development with Watcom compiler, if it compiles Birthright?

Probably, if you want to continue development of this game, then first tasks would be :
1) Add support of more modern DirectX : 9 - 12.
2) Move rendering from software rendering to GPU.
3) Increase resolution of textures and make them 32 bits.
4) Increase game rendering resolutions to current Full HD, 2K and 4K.

**Comment by Shiryou** [comment]
@RodionKarimov There's work being done to get the original development environment figured out in juanitogan/rbxit#13

The more experienced eyes we can get on it, the better.
